### ISSUE TITLE
What still needs to be done?

### ISSUE THREAD

**Issue by nwilder0** [claytondus/cdnwsh#27]
@claytondus 
I'm continuing to add cmds and error handling to the shell wrapper functions.  And I will add client functionality next and then add more thorough testing for both server and client.

Do you still need me to pick up part of directory implementation?

I will also go ahead and implement the cmd line script of commands functionality

**Comment by claytondus** [comment]
Great.  I'm working on mkdir/rmdir right now, so I got the directories covered.  I'll try to put in everything else I can think of as an issue right now.

**Comment by nwilder0** [comment]
FYI, the select() call kind of flaked out on me for a few hours, I found a workaround finally, but I need to clean up the code before its working and commitable.  So, I'm going to pick up first thing in the morning.

**Comment by claytondus** [comment]
OK, I'm still working on directory traversal.

**Comment by claytondus** [comment]
Dir traversal is done.  Files coming next.  All hands on deck tomorrow.

**Comment by nwilder0** [comment]
Yep, I'm working tonight as well...I finally gave up on select(), too many bugs, just switched everything to use poll() instead

**Comment by claytondus** [comment]
This may be too late to suggest, but libevent or libuv is a good abstraction over those.

**Comment by nwilder0** [comment]
Yeah, well, all seems to be working well with poll() so hopefully it won't have to be touched again

**Comment by claytondus** [comment]
Before I do tree, I'm going to make single indirect blocks work.  After that, I'll start in on the document.  I have a conference call at 7:30, should take 30 min or so.  Is everything with the shell and remote working?

**Comment by nwilder0** [comment]
For the most part everything seems to be working.  'ls' is outputting some non-text characters at the start of the directory listing after doing a mkdir, although eveything else about it seems fine.  I'll continue looking into it.  Also, it might be helpful to have a function that returns the size of a directory listing or cat output as right now I'm just creating a 4K buffer.  But it shouldn't be too difficult to add, I would think.
I'll continue trying to find bugs and test things. 

```
On Monday, February 29, 2016 5:37 PM, Clayton Davis <notifications@github.com> wrote:
```

 Before I do tree, I'm going to make single indirect blocks work. After that, I'll start in on the document. I have a conference call at 7:30, should take 30 min or so. Is everything with the shell and remote working?—

**Comment by claytondus** [comment]
The ls thing you can take care of by memset'ing the buffer before you call cnls(), look at the test LsShouldComplete in test_fs.c. The size of a directory or file is available through cnstat and then reading the inode, or by looking at cwd->inode_st, although they won't grow bigger than 4K (because the directory won't save more than one block right now).  Thanks.
